src/em/prov/easyconnect/ec_pa_configurator.cpp

```diff
@@ -263,3 +287,114 @@ bool ec_pa_configurator_t::process_proxy_encap_dpp_msg(em_encap_dpp_t *encap_tlv
     free(encap_frame);
     return did_finish;
 }
+
+bool ec_pa_configurator_t::handle_gas_comeback_request([[maybe_unused]] uint8_t *buff, [[maybe_unused]] unsigned int len, uint8_t sa[ETH_ALEN])
+{
+    em_printfout("Received a GAS Comeback Request frame from '" MACSTRFMT "'", MAC2STR(sa));
+    m_peer_ready_for_frag_frame[util::mac_to_string(sa)] = true;
+    auto frame_it = m_gas_frames_to_be_sent.find(util::mac_to_string(sa));
+    if (frame_it == m_gas_frames_to_be_sent.end()) {
+        // Nothing to do
+        em_printfout("Received potentially spurious GAS Comeback Request frame from '" MACSTRFMT "', not doing anything with it", MAC2STR(sa));
+        return true;
+    }
+
+    // Ensure we already have a GAS session with this peer
+    auto dialog_it = m_gas_session_dialog_tokens.find(util::mac_to_string(sa));
+    if (dialog_it == m_gas_session_dialog_tokens.end()) {
+        em_printfout("Received GAS Comeback Request from '" MACSTRFMT "', we have a frame waiting for them, but no dialog token known!", MAC2STR(sa));
+        return false;
+    }
+
+    // Get the fragments to send
+    std::vector<ec_gas_comeback_response_frame_t*>& fragments = frame_it->second;
+
+    if (fragments.empty()) {
+        em_printfout("Received GAS Comeback Request, but we have no more fragments to send to '" MACSTRFMT "'", MAC2STR(sa));
+        return false;
+    }
+
+    // This code assumes that m_gas_frames_to_be_sent stores fragments with linearly increasing frag ID
+    // Grab 0th, send it, delete it
+    ec_gas_comeback_response_frame_t *fragment = fragments.front();
+    fragments.erase(fragments.begin());
+
+    bool sent = m_send_action_frame(sa, reinterpret_cast<uint8_t*>(fragment), sizeof(ec_gas_comeback_response_frame_t) + fragment->comeback_resp_len, 0, 0);
+    
+    if (!sent) {
+        em_printfout("Failed to send fragment #%d to '" MACSTRFMT "'", fragment->fragment_id, MAC2STR(sa));
+    }
+
+    em_printfout("Sent fragment #%d (more frags = %d) to '" MACSTRFMT "'", fragment->fragment_id, fragment->more_fragments, MAC2STR(sa));
+    free(fragment);
+    return sent;
+}
+
+bool ec_pa_configurator_t::send_prepare_for_fragmented_frames_frame(uint8_t dest_mac[ETH_ALEN])
+{
+    auto it = m_gas_session_dialog_tokens.find(util::mac_to_string(dest_mac));
+    if (it == m_gas_session_dialog_tokens.end()) {
+        em_printfout("No GAS session dialog token found for '" MACSTRFMT "', cannot send fragmented frame!",  MAC2STR(dest_mac));
```

Comment:
```suggestion
        em_printfout("No GAS session dialog token found for '" MACSTRFMT "', cannot send fragmentation indication message to peer!",  MAC2STR(dest_mac));
```
(or something better)